langchain/src/vectorstores/vectara.ts

```diff
@@ -361,37 +516,38 @@ export class VectaraStore extends VectorStore {
   }
 
   /**
-   * Performs a similarity search and returns documents along with their
-   * scores.
+   * Performs a Vectara API call based on the arguments provided.
    * @param query The query string for the similarity search.
    * @param k Optional. The number of results to return. Default is 10.
    * @param filter Optional. A VectaraFilter object to refine the search results.
    * @returns A Promise that resolves to an array of tuples, each containing a Document and its score.
    */
-  async similaritySearchWithScore(
+  async vectara_query(
     query: string,
-    k = 10,
-    filter: VectaraFilter | undefined = undefined
-  ): Promise<[Document, number][]> {
+    k: number,
+    _filter: VectaraFilter,
+    _summary: VectaraSummary = { enabled: false, maxSummarizedResults: 0, responseLang: "eng" },
+  ): Promise<SummaryResult> {
     const headers = await this.getJsonHeader();
+    const { start, filter, lambda, contextConfig, mmrConfig } = _filter;
 
     const corpusKeys = this.corpusId.map((corpusId) => ({
       customerId: this.customerId,
       corpusId,
-      metadataFilter: filter?.filter ?? "",
-      lexicalInterpolationConfig: { lambda: filter?.lambda ?? 0.025 },
+      metadataFilter: filter,
+      lexicalInterpolationConfig: { lambda },
     }));
 
     const data = {
       query: [
         {
           query,
-          numResults: k,
-          contextConfig: {
-            sentencesAfter: filter?.contextConfig?.sentencesAfter ?? 2,
-            sentencesBefore: filter?.contextConfig?.sentencesBefore ?? 2,
-          },
+          start,
+          numResults: mmrConfig?.enabled ? mmrConfig.mmrTopK : k,
```

Comment:
I think a lot of this would ideally be handled by the retriever created from `vectorstore.asRetriever()`? Or is it that we want this retriever to be able to do everything?

Reply:
See the new version. Separated VectaraStore from VectaraSummaryRetriever in a different file. For this comment, do you mean that the vectaraQuery() function belongs more in the retriever than in the store? (and store can call it in e.g. similarity_search as a retriever?)
If so, would the constructor generate the retriever inside the VectaraStore?